### PR TITLE
User can revisit the page to continue sketching on the same canvas. 

### PR DESCRIPTION
**Pull request by danielduffield**
Created a system to save all canvas data by combining period canvas save data with unsavedData collected into a set of actions to be sent and executed upon connection to users.

### CODE REVIEW

**Comment by danielduffield** [comment]
https://con-canvas.herokuapp.com/

**Comment by danielduffield** [comment]
concanvas-issue-3

File: canvas-state/blank.json [hunk 1]
```diff
@@ -0,0 +1 @@
...AgQIBALCCwYlBzBAgQIECAAAGB5QcIECBAgAABArGAwIpBzREgQIAAAQIEBJYfIECAAAECBAjEAgIrBjVHgAABAgQIEBBYfoAAAQIECBAgEAsIrBjUHAECBAgQIEBAYPkBAgQIECBAgEAsILBiUHMECBAgQIAAAYHlBwgQIECAAAECsYDAikHNESBAgAABAgQElh8gQIAAAQIECMQCAisGNUeAAAECBAgQEFh+gAABAgQIECAQCwisGNQcAQIECBAgQEBg+QECBAgQIECAQCwgsGJQcwQIECBAgAABgeUHCBAgQIAAAQKxgMCKQc0RIECAAAECBASWHyBAgAABAgQIxAICKwY1R4AAAQIECBAQWH6AAAECBAgQIBALCKwY1BwBAgQIECBAQGD5AQIECBAgQIBALCCwYlBzBAgQIECAAAGB5QcIECBAgAABArGAwIpBzREgQIAAAQIEBJYfIECAAAECBAjEAgIrBjVHgAABAgQIEBBYfoAAAQIECBAgEAsIrBjUHAECBAgQIEBAYPkBAgQIECBAgEAsILBiUHMECBAgQIAAAYHlBwgQIECAAAECsYDAikHNESBAgAABAgQElh8gQIAAAQIECMQCAisGNUeAAAECBAgQEFh+gAABAgQIECAQCwisGNQcAQIECBAgQEBg+QECBAgQIECAQCwgsGJQcwQIECBAgAABgeUHCBAgQIAAAQKxgMCKQc0RIECAAAECBASWHyBAgAABAgQIxAICKwY1R4AAAQIECBAQWH6AAAECBAgQIBALCKwY1BwBAgQIECBAQGD5AQIECBAgQIBALCCwYlBzBAgQIECAAAGB5QcIECBAgAABArGAwIpBzREgQIAAAQIEBJYfIECAAAECBAjEAgIrBjVHgAABAgQIEBBYfoAAAQIECBAgEAsIrBjUHAECBAgQIEBAYPkBAgQIECBAgEAsILBiUHMECBAgQIAAAYHlBwgQIECAAAECsYDAikHNESBAgAABAgQElh8gQIAAAQIECMQCAisGNUeAAAECBAgQEFh+gAABAgQIECAQCwisGNQcAQIECBAgQEBg+QECBAgQIECAQCwgsGJQcwQIECBAgAABgeUHCBAgQIAAAQKxgMCKQc0RIECAAAECBASWHyBAgAABAgQIxAICKwY1R4AAAQIECBB4CI8CWV89EyAAAAAASUVORK5CYII="}
```
Comment (thebearingedge) [inline canvas-state/blank.json:1]: Did you mean to check this in?

File: client/utils/canvas.js [hunk 2]
```diff
@@ -6,27 +6,66 @@ const socket = io.connect()
...
-  paintEvent(mouseX, mouseY, previousX, previousY) {
+  async loadCanvas() {
     this.ctx = this.canvas.getContext('2d')
     this.ctx.fillStyle = '#000000'
+    const img = new Image()
+    img.onload = () => {
+      this.ctx.drawImage(img, 0, 0)
+    }
+    const response = await fetch('/saved-canvas')
+    const canvasData = JSON.parse(await response.json())
```
Comment (thebearingedge) [inline client/utils/canvas.js]: Are you sure you need to `parse` here? `reponse.json()` should do the parsing for you.

File: server/utils/fsAsyncFunctions.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
+const fs = require('fs')
```
Comment (thebearingedge) [inline server/utils/fsAsyncFunctions.js]: Check out `fs-promise` or Node 8's `util.promisify`

**Review by thebearingedge — CHANGES_REQUESTED** [review]

**Review by thebearingedge — APPROVED** [review]

**Pull request merged**